File: solutions/heaps/kthLargestElementInAnArray.js
// https://www.npmjs.com/package/heap-js
const heapJS = require('heap-js');

const findKthLargest = (nums, k) => {
    const kLargest = new heapJS.Heap();
    
    for (const num of nums) {
        kLargest.push(num);
        if (kLargest.size() > k) {
            kLargest.pop();
        }
    }
    
    const kthLargest = kLargest.pop();
    return kthLargest;
};

File: solutions/python/heaps/kth_largest_element_in_an_array.py
import heapq

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        kth_largest_elements = []

        for num in nums:
            heapq.heappush(kth_largest_elements, num)

            if len(kth_largest_elements) > k:
                heapq.heappop(kth_largest_elements)

        kth_largest_element = heapq.heappop(kth_largest_elements)
        return kth_largest_element